src/studio/src/designer/frontend/packages/schema-editor/src/components/SchemaInspector.tsx

```diff
@@ -98,6 +98,22 @@ export const isValidName = (name: string) => {
   return Boolean(name.match(/^[a-zA-ZæÆøØåÅ][a-zA-Z0-9_.\-æÆøØåÅ ]*$/));
 };
 
+export const isNameInUse = (params: { parentSchema: UiSchemaItem | null, path: string, name: string }) => {
```

Comment:
I would simplify this to just destruct the parameters directly, and remove the `params`. IMO it makes it a bit easier to read, and avoids a lot of repetition - but might just be my preference :)

The checks can also be simplified a bit to avoid the many return statments, which can be confusing I think.

```typescript
const isNameInUse = ({ parentSchema: UiSchemaItem | null, path: string, name: string }) => {
  // Check if the parent node has other children with the same name.
  if (
    parentSchema?.properties.some(
      (p) => p.displayName === name && p.path !== path
    )
  ) {
    return true;
  }
  
  return false;
};
```

Reply:
I'm just curious here and haven't looked at how the `parentSchema` is fetched. But does this also cover the case for items on root which don't actually have a parent schema?

Reply:
Good point @lorang92 - seems like this is thightly connected to the use of the properties keyword (as everything else in this area) and hence this change doesn't make things worse in that regards, but I'll see if I can improve it to future proof it so we don't have to revisit this when we make that bigger change.